library/Exceptions/NestedValidationException.php

```diff
@@ -22,76 +22,75 @@
 use function is_array;
 use function str_repeat;
 
+/**
+ * Exception for nested validations.
+ *
+ * This exception allows to have exceptions inside itself and providers methods
+ * to handle them and to retrieve nested messages based on itself and its
+ * children.
+ *
+ * @author Alexandre Gomes Gaigalas <alexandre@gaigalas.net>
+ * @author Henrique Moody <henriquemoody@gmail.com>
+ * @author Jonathan Stewmon <jstewmon@rmn.com>
+ * @author Wojciech Frącz <fraczwojciech@gmail.com>
+ */
 class NestedValidationException extends ValidationException implements IteratorAggregate
 {
     /**
      * @var SplObjectStorage
      */
-    private $exceptions = [];
+    private $exceptions;
 
     /**
-     * @param ValidationException $exception
+     * Returns the exceptions that are children of the exception.
      *
-     * @return self
+     * @return SplObjectStorage|ValidationException[]
      */
-    public function addRelated(ValidationException $exception)
+    public function getRelated(): SplObjectStorage
     {
-        $this->getRelated()->attach($exception);
+        if (!$this->exceptions instanceof SplObjectStorage) {
+            $this->exceptions = new SplObjectStorage();
```

Comment:
what about initialize it in the constructor?

Reply:
I'd rather keep it for now. I wanna get a rid of `SplObjectStorage` soon because it allows users to add any objects to the stack